### ISSUE TITLE
Webservice Basic Auth not working

### ISSUE THREAD

**Issue by chokdee** [citrusframework/citrus#423]
I have tried to setup the basic authentification for the webservice client. But it doesn't seems to work

```
<bean id="basicAuthClient" class="org.springframework.ws.transport.http.HttpComponentsMessageSender">
      <property name="authScope">
         <bean class="org.apache.http.auth.AuthScope">
            <constructor-arg value="localhost"/>
            <constructor-arg value="6666"/>
            <constructor-arg value=""/>
            <constructor-arg value="basic"/>
         </bean>
      </property>
      <property name="credentials">
         <bean class="org.apache.http.auth.UsernamePasswordCredentials">
            <constructor-arg value="username"/>
            <constructor-arg value="password"/>
         </bean>
      </property>
   </bean>

   <citrus-ws:client id="webServiceClient"
                     timeout="1000"
                     message-sender="basicAuthClient"
                     request-url="http://localhost:6666/endpoint"/>
```

Gives me a 401.
The request log on the server do not show the username so it isn't send.

If I try it like this:
```
<send>
...
<header><element name="citrus_http_Authorization" value="citrus:concat('Basic ', citrus:encodeBase64('username:password'))"/></header>

```
it's working

**Comment by svettwer** [comment]
Hi @chokdee !
I traced down the issue to the line where the request is sent.
The `WebServiceMessageSender` is configured in the `WebServiceTemplate` as it should.

I also checked the [documentation of spring ws](https://docs.spring.io/spring-ws/docs/2.4.2.RELEASE/reference/#_http_transports) just to be sure that the API is used correctly.

It seems that the issue is related to the `AuthScope` bean. 
What you've in place (and what the documentation states) is:
```xml
<bean class="org.apache.http.auth.AuthScope">
  <constructor-arg value="localhost"/>
  <constructor-arg value="6666"/>
  <constructor-arg value=""/>
  <constructor-arg value="basic"/>
</bean>
```
This leads to a configuration, where the realm is set to `""` instead of being applied to any realm. If you want to authenticate against any realm, you should use the following config.
```xml
<bean class="org.apache.http.auth.AuthScope">
  <constructor-arg value="localhost"/>
  <constructor-arg value="6666"/>
  <constructor-arg><null /></constructor-arg>
  <constructor-arg value="basic"/>
</bean>
```
This bean injects a null value for the `realm` property into the `AuthScope` constructor and enables the authentication against any realm.

If you don't need a scope at all, you could just drop the whole `AuthScope` bean. This will enable the authentication against any endpoint.

Nevertheless, one should alter the documentation so that it does not lead to this issue.

BR,
Sven

**Comment by svettwer** [comment]
Merged #433 with updated documentation.
